index.bs

```diff
@@ -6145,6 +6244,354 @@ However, [=authenticators=] that do not utilize [[!FIDO-CTAP]] do not necessaril
+        ;
+        ; Whatever the scope, a device key is still specific to a given credential
+        ; and does not provide any ability to link credentials.
+        ;
+        ; Whether device-scoped or not, keys are still device-bound. I.e. an
+        ; app-scoped key does not enjoy lesser protection from extraction.
+
+        scope: uint .size 1, ; a value of 0x00 means "entire device" ("all apps") scope.
+                             ; 0x01 means "per-app" scope.
+                             ; Values other than 0x00 or 0x01 are reserved for future
+                             ; use.
+
+        ; An authenticator-generated random nonce for inclusion in the attestation
+        ; signature. If the authenticator chooses to not generate a nonce, it sets this
+        ; to a zero-length bytestring.
+
+        nonce: bstr .size (0..32),
+
+        ; See https://www.w3.org/TR/webauthn/#sctn-generating-an-attestation-object
+        ;
+        ; Attestation statement formats define the \`fmt\` and \`attStmt\` members of
+        ; $$attStmtType.
+        ;
+        ; In summary, the \`attStmt\` will (typically) contain:
+        ;   (1) a SIGNATURE value calculated (using the attestation private key)
+        ;       over (prefix || aaguid || dpk || nonce) where \`prefix\` is
+        ;       h'64657669636520626f756e64206b6579206174746573746174696f6e20736967
+        ;         00ffffffff'
+        ;   (2) the attestation certificate or public key, and supporting certificates,
+        ;       if any.
+        ;
+        ; Note that there are details dependent upon the particular attestation
+        ; statement format.
+        ; See https://www.w3.org/TR/webauthn/#sctn-defined-attestation-formats.
+
+        $$attStmtType,
+
+        ; An optional boolean that indicates whether the attestation statement
+        ; contains uniquely identifying information. This can only be true
+        ; when the \`attestation\` field of the extension input is "enterprise"
+        ; an either the user-agent or the authenticator permits uniquely
+        ; identifying attestation for the requested RP ID.
+
+        ? epAtt: bool .default false,
+    }
+
+    ```
+
+: Unsigned extension output
+:: A signature generated with the device public key as a CBOR bytestring
+
+: Authenticator extension processing
+:: For both [=authenticatorMakeCredential=] and [=authenticatorGetAssertion=] operations:
+    1. Create or select the [=public key credential source=] as usual (see [[#sctn-op-make-cred]], or [[#sctn-op-get-assertion]] as appropriate).
+
+    1. If a [=hardware-bound device key pair=] does not already exist for this {[=public key credential source/id|Credential ID=], [=public key credential source/rpId|RP ID=], [=public key credential source/rpId|userHandle=]} tuple on the [=authenticator=], create it using the same public key algorithm as that used by the [=user credential=]'s [=credential key pair=], otherwise locate the existing [=device-bound key=].
```

Comment:
Why the same public key algorithm? e.g. what happens if a primary credential uses -7, and then DPK is requested on a device that has a hardware TPM or similar which doesn't support ES256? Is that too fictional?

Reply:
There's also the perspective that `pubKeyCredParams` lists the algorithms the RP is capable of verifying, so by keeping the dpk restricted to the same algorithms we ensure that if the RP can verify the primary signature, it can also verify the dpk signature.